src/main/resources/com/openshift/jenkins/plugins/OpenShiftDSL.groovy

```diff
@@ -808,12 +808,11 @@ class OpenShiftDSL implements Serializable {
         return r;
     }
 
-    public Result delete(Object... oargs) {
-        String[] args = toStringArray(oargs);
-        Result r = new Result("delete");
-        r.actions.add((OcAction.OcActionResult)script._OcAction(buildCommonArgs("delete", null, args)));
-        r.failIf("delete returned an error");
-        return r;
+    public Result delete(Object obj,Object... args) {
```

Comment:
i'm confused by the implications of this signature change on existing callers of delete()?

i trust you but i wouldn't mind an explanation of what has actually happened here that we can introduce the new arg safely.  It looks like previously the first arg was implicitly always the "resource type" and now that is explicit?

Reply:
Certainly / no worries ... and I wasn't sure until I verified with some test cases, but as I lay this out, I realize an additional scenario should be tried ... perhaps I did break something...

First, why I did it?  It made it consistent with the create, and it made it easier to isolate the first arg, which `objectDefAction` needed.... and dealing with `Object...` in groovy has proven a pain

Next, why it works:
a) `objDefAction` handles the different cases ... i.e. is the first art a resource type? is the first arg a list of api objs (which are in either groovy/map or json format) ?
b) maybe java is too, groovy seems to be very permissive wrt Object[] and Object....  being combined, omitted 

Now, all that said, I do think I need to go back and try more things like `openshift.delete("svc/mongodb", "dc/mongodb") ... I thought I tried that and it worked (i.e. it was existing), but I don't see it now, and I didn't add it here.

I'll do that (may wait until Monday) and report back.

Reply:
Reporting back with some findings .... the plus side is I get to highlight the shortcomings of groovy again :-)

So, this works in Java:

```
    public void delete(Object... args) {
    }

    public void delete(Object obj,Object... args) {
    }
``` 

But not in groovy ... keeping the original delete signature, and adding a new signature, such as:

```
    public Result delete(Object... args) {
        logToTaskListener("GGM original signature ");
        String[] sarr = toStringArray(args);
        ArrayList<String> al = Arrays.asList(sarr);
        logToTaskListener("GGM with params " + al);
        Object firstObj = al.remove(0);
        return delete(firstObj, al.toArray());
    }

    public Result delete(Object obj,Object... args) {
        logToTaskListener("GGM new signature " + obj)
        OpenShiftResourceSelector deleteSelector = objectDefAction("delete", obj, args);
        //NOTE: more groovy suckage ... I could not pass a OpenShiftResourceSelector, even
        // though it extends Result, into the Result(Result) contructor
        return new Result(deleteSelector.highLevelOperation, deleteSelector.actions);
    }

```

Results in this complaint:

```
java.lang.IllegalStateException: cannot compare public com.openshift.jenkins.plugins.OpenShiftDSL$Result com.openshift.jenkins.plugins.OpenShiftDSL.delete(java.lang.Object[]) to public com.openshift.jenkins.plugins.OpenShiftDSL$Result com.openshift.jenkins.plugins.OpenShiftDSL.delete(java.lang.Object,java.lang.Object[])
	at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.GroovyCallSiteSelector.isMoreSpecific(GroovyCallSiteSelector.java:291)
	at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.GroovyCallSiteSelector.findMatchingMethod(GroovyCallSiteSelector.java:203)
	at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.GroovyCallSiteSelector.method(GroovyCallSiteSelector.java:150)
	at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.SandboxInterceptor.onMethodCall(SandboxInterceptor.java:87)
	at org.kohsuke.groovy.sandbox.impl.Checker$1.call(Checker.java:155)
	at org.kohsuke.groovy.sandbox.impl.Checker.checkedCall(Checker.java:159)
	at com.cloudbees.groovy.cps.sandbox.SandboxInvoker.methodCall(SandboxInvoker.java:17)
	at WorkflowScript.run(WorkflowScript:5)
```

So this tells me changing the signature like I have done above does not break things.  Existing invocations will reach the innards of our new method like before.

Lastly, this test, which I'll add, works:

```
        openshift.create("configmap", "foo")
        openshift.create("configmap", "bar")
        openshift.delete("configmap/foo", "configmap/bar")
```

Temporary debug shows that' `configmap/foo` is in the first param, and `configmap/bar` is the first element in the args array.

I'll upload the updated test case file in a bit to add that snippet.  I also need to update the global var help file to highlight the new capabilities for `openshift.delete`